src/server/blockchain/AdminWallet.js

```diff
@@ -600,6 +649,95 @@ export class Wallet {
       throw new Error(e)
     }
   }
+
+  /**
+   * Helper function to handle a tx Send call
+   * @param tx
+   * @param {object} promiEvents
+   * @param {function} promiEvents.onTransactionHash
+   * @param {function} promiEvents.onReceipt
+   * @param {function} promiEvents.onConfirmation
+   * @param {function} promiEvents.onError
+   * @param {object} gasValues
+   * @param {number} gasValues.gas
+   * @param {number} gasValues.gasPrice
+   * @returns {Promise<Promise|Q.Promise<any>|Promise<*>|Promise<*>|Promise<*>|*>}
+   */
+  async sendTransactionMainnet(
+    tx: any,
+    txCallbacks: PromiEvents = {},
+    { gas, gasPrice }: GasValues = { gas: undefined, gasPrice: undefined }
+  ) {
+    let currentAddress
+    try {
+      const { onTransactionHash, onReceipt, onConfirmation, onError } = txCallbacks
+      gas =
+        gas ||
+        (await tx
+          .estimateGas()
+          .then(gas => gas + 50000) //buffer for proxy contract, reimburseGas?
```

Comment:
i think better will be to move those constants somewhere to the config and add corresponding env vars